apps/api/src/models/post.ts

```diff
@@ -0,0 +1,50 @@
+// Node JS approach
+// export type Post = {
+//   id: string;
+//   title: string;
+//   image: string;
+//   description: string;
+//   category: string;
+//   comments: Array<[]>;
+// };
+
+// MongoDB approach
+import mongoose, { Document, Schema, Types } from 'mongoose';
+
+interface IPost extends Document {
+  // id: string;
+  title: string;
+  image: string;
+  description: string;
+  category: string;
+  comments?: Types.Array<String>;
+}
+
+export const postSchema = new Schema<IPost>({
+  // id: {
+  //   type: String,
+  //   required: true
+  // },
+  title: {
+    type: String,
+    required: [true, 'Property is required']
+  },
+  image: {
+    type: String
+  },
+  description: {
+    type: String,
+    required: [true, 'Property is required']
+  },
+  category: {
```

Comment:
Category can be a reference to another document, so you just save the ID in your post, it will look something like this
```
category: {
      type: Schema.Types.ObjectId,
      ref: 'Category',
      required: [true, 'Property is required']
    }
```